src/hardware/pcspeaker.cpp

```diff
@@ -527,10 +514,10 @@ static void PCSPEAKER_CallBack(Bitu len)
 	// shift out consumed samples
 	// TODO: use ring buffer or something to avoid shifting
 	for (unsigned i = len; i < output_buffer_length; i++) {
-		NASSERT(i - len < 0, "index into output_buffer too low");
-		ASSERT(i - len < output_buffer_length,
-		       "index into output_buffer too high");
-		ASSERT(i < output_buffer_length, "index into output_buffer too high");
+		assertm(i - len > 0, "index into output_buffer too low");
```

Comment:
You flipped the direction of this check.

Reply:
Yeah, I didn't like the prior assert, so decided invert the logic - but it's still not at all intuitive. Will get this fixed up.

Reply:
Resolving; check is replaced with array `.at(index)`